database/memory/db.go

```diff
@@ -375,12 +398,53 @@ func (db *MemoryDB) GetEventsFromAddressTotal(address types.Address, options *ty
 	return uint64(len(db.eventIndexDB[address])), nil
 }
 
-func (db *MemoryDB) GetStorageWithOptions(types.Address, *types.PageOptions) ([]*types.StorageResult, error) {
-	return nil, database.ErrNotImplemented
+func (db *MemoryDB) GetStorageWithOptions(address types.Address, options *types.PageOptions) ([]*types.StorageResult, error) {
+	db.mux.RLock()
+	defer db.mux.RUnlock()
+	if !db.addressIsRegistered(address) {
+		return nil, errors.New("address is not registered")
+	}
+	var convertedList []*types.StorageResult
+
+	fromBlockNum := options.BeginBlockNumber.Uint64()
+	endBlockNum := options.EndBlockNumber.Int64()
+	blockNum := fromBlockNum
+
+	storageIndexer, ok := db.storageIndexDB[address]
+	if ok {
+		for blkNum, storageRoot := range storageIndexer.root {
+			if blkNum >= fromBlockNum && (blkNum <= uint64(endBlockNum) || endBlockNum == -1) {
+				convertedList = append(convertedList, &types.StorageResult{
+					Storage:     storageIndexer.storage[storageRoot],
+					StorageRoot: types.NewHash(storageRoot),
+					BlockNumber: blockNum,
```

Comment:
```suggestion
					BlockNumber: blkNum,
```
Wrong variable used, `blockNum` is not used so remains 0.

Reply:
done